### PR TITLE
CMake: Set the tool configuration for osx, z/os.

### PR DESCRIPTION
**Pull request by rwy0717**
The z/os configuration is wrong, but at least we'll try something.

/cc @dnakamura @mgaudet @charliegracie 
Signed-off-by: Robert Young <rwy0717@gmail.com>

### CODE REVIEW

**Comment by rwy0717** [comment]
Rebased and fixed conflict.

File: cmake/modules/OmrDetectSystemInformation.cmake [hunk 1]
```diff
@@ -83,11 +83,13 @@ macro(omr_detect_system_information)
 
    if(OMR_HOST_OS STREQUAL "zos")
       # ZOS specifics
+      set(OMR_TOOLCONFIG "gnu")
    endif()
 
 
    if(OMR_HOST_OS STREQUAL "osx")
       # OSX specifics
+      set(OMR_TOOLCONFIG "gnu")
```
Comment (charliegracie) [inline cmake/modules/OmrDetectSystemInformation.cmake]: Do we want to create a clang tool config?

Comment (rwy0717) [inline cmake/modules/OmrDetectSystemInformation.cmake]: I believe the long-run intention is that gnu will cover all gnu-ish toolchains, including clang. This is just the smallest change I could make to get osx/cmake building.

**Comment by rwy0717** [comment]
This is needed to enable travis builds in #1475. Check out the failed build [here](https://travis-ci.org/rwy0717/omr/jobs/262567501). We fail to set the toolchain and `exit 1`.

File: cmake/modules/OmrDetectSystemInformation.cmake [hunk 2]
```diff
@@ -83,11 +83,13 @@ macro(omr_detect_system_information)
 
    if(OMR_HOST_OS STREQUAL "zos")
       # ZOS specifics
+      set(OMR_TOOLCONFIG "gnu")
```
Comment (rwy0717) [inline cmake/modules/OmrDetectSystemInformation.cmake]: yeah I wrote in the commit, this setting is wrong, but at least there's something here while we work on proper xlc support.

Comment (charliegracie) [inline cmake/modules/OmrDetectSystemInformation.cmake]: I am ok with this for now so that we can move forward

**Pull request merged**